### DISCUSSION TITLE
PublicKey is not generated as expected Coin.EDDSA

### DISCUSSION THREAD

**Discussion by Sotatek-ToanTran** [CoolBitX-Technology/coolwallet-js-sdk#271] [Q&A] [answered]
# PublicKey is not generated as expected Coin.EDDSA

- Description
We're trying generate pubkey from this.getPublicKey() to get Address with Coin.EDDSA. We tried testing on XLM chain folder code but it doesn't seem to work. (XLM is also the only that uses EDDSA in coolwallet-sdk).

This is a code common to get publicKey from this.publicKey(), it's not expected.
```js
// Get Address EDDSA
async getAddress(transport: types.Transport, appPrivateKey: string, appId: string, addressIndex: number): Promise<string> {
  const publicKey = await this.getPublicKey(transport, appPrivateKey, appId, addressIndex);
  // publicKey output: 7d6e279e17a99cc2af1c9aacd18ef626b839369abd57672bff5c7fc878ce0952
  // return txUtil.pubKeyToAddress(publicKey, this.addressType);
}

// *Expected publicKey*:
// BIP 44, addressIndex 0: 74d0b6cc6a67c4041a8ab2611f054a9ca57fa1c6e4b29178bfe9462d49c8fbf9
// BIP-44 Change, addressIndex 0: 910a5ec043d8e79a6eb11d45d2fb6a2e34ef141f487ad5970f77a2b0cf24c4fe
```

- Expected Behavior
How is the publicKey generated using this.getPublicKey()? 
> We need clear this logic generated a publickey from card to convert valid publickey with SOL chain! Each chain, you all use different transformations, we need to understand the initial publicKey from this.getPublicKey().

- Screenshort Behavior
### XLM Error when run
![image](https://user-images.githubusercontent.com/31679583/152969574-9ad17e0c-467a-4927-b946-ecabfcd34bf8.png)

**Comment by hank121314** [comment, marked as answer]
Sorry for this issue.
We have fixed it in the master branch. Please pull the latest change in master branch and test it again!

Thanks!

Reference: https://github.com/CoolBitX-Technology/coolwallet-sdk/pull/272
